### ISSUE TITLE
Shiny app for the trial conduct part

### ISSUE THREAD

**Issue by danielinteractive** [Roche/crmPack#479]
Idea: 
- Running simulations etc. might not be the best spot for a Shiny app, because computations just take too long
- However, once the design is fixed, for conducting the dose escalation meetings etc. it could be very helpful
- From a concept point of view, it could be similar as with teal apps where the app developer / statistician locks in the design and then the UI in the app allows to add different cohort data etc.

**Comment by PuzzledFace** [comment]
The app could define the designs (and truths) to be simulated and then schedule the batch job to run and report on them.  I think a Shiny app that could do that would still be helpful.  

This is something I'd like to be involved it.

**Comment by danielinteractive** [comment]
Thanks John for your initiative on this. Regarding simulations, I think it is a bit early to start coding on this now, since we should first improve the HPC compatibility of `crmPack` in general, have a static template that shows how to do this. Afterwards it could then be interesting to prototype a simple app that helps with executing this template.
So good to start thinking about it, but I think it would be better to get these other pieces done first. 
What do you think?

**Comment by PuzzledFace** [comment]
> Thanks John for your initiative on this. Regarding simulations, I think it is a bit early to start coding on this now, since we should first improve the HPC compatibility of `crmPack` in general, have a static template that shows how to do this. Afterwards it could then be interesting to prototype a simple app that helps with executing this template. So good to start thinking about it, but I think it would be better to get these other pieces done first. What do you think?

I agree that simulation should be the final part of the puzzle, and that the fine details of the app should wait until the current major refactoring is complete.  However, there is much that can be done in the mean time.  The introspection that will be need to allow the app's GUI to be independent of `crmPack` (and any local user extensions) will take some time to implement.  And we will need a similar customisable interface for reporting as well.  (That ties in neatly with something I'm doing elsewhere already.)  So there is plenty to do in the mean time...

Some details to ponder:
-  What should the name of the app be?  I used shinyCRM, but shinyCRMPack might be more appropriate (though the capitalisation is awkward).
- Should the app repo be on github.com/Roche or github.com/openpharma or elsewhere?
- If we decide to host the app on our own webserver, where will funding come from?

**Comment by danielinteractive** [comment]
Thanks John - yeah for sure design work can start before, no problem. As a principle it would be perfect to first close out the ongoing issues and PRs, so that we get to a "one person - one issue ongoing" stage :-) (Let me know if any blockers on the ongoing stuff)
- name could be `shinyCrmPack` e.g. (no need to upper case the whole CRM acronym I think)
- repo ideally on github.com/openpharma
- I would recommend to think about a "teal" way of doing this, i.e. make it Shiny modules that can easily be combined and spun up into an app hosted where anybody wants to host it. That way, the app developer can hardcode things easily, which the users don't need to select anymore etc. Then the webserver question is not important anymore, we could just host a free shiny.io app somewhere for demo purposes.
